src/compiler/crystal/tools/doc/method.cr

```diff
@@ -177,10 +177,22 @@ class Crystal::Doc::Method
     !@def.args.empty? || @def.block_arg || @def.yields
   end
 
+  def args_to_json
+    @def.args.map_with_index do |arg, i|
+      {
+        name:         arg.external_name,
+        type:         arg.type?.to_s,
+        default:      arg.default_value.to_s,
+        splat:        @def.splat_index == i,
+        double_splat: @def.double_splat ? true : false,
```

Comment:
I've been thinking, `splat ^ double_splat`, so perhaps this should just be a single attribute with `splat` with values `none`, `single`, `double`? What would be easier to consume in a program?